File: ir/types/label.go
package types

import (
	"io"

	"github.com/panda-io/micro-panda/ir/core"
)

type LabelType struct {
}

func (t *LabelType) Equal(u core.Type) bool {
	if _, ok := u.(*LabelType); ok {
		return true
	}
	return false
}

func (t *LabelType) String() string {
	return "label"
}

func (t *LabelType) WriteIR(w io.Writer) error {
	_, err := w.Write([]byte("label"))
	return err
}

File: app/src/main/java/com/github/panda_io/micro_panda/builder/llvm/ir/type/Label.java
package com.github.panda_io.micro_panda.builder.llvm.ir.type;

public class Label extends Type {
    public boolean equal(Type type) {
        return type instanceof Label;
    }

    public String string() {
        return "label";
    }

    public void writeIR(StringBuilder builder) {
        builder.append("label");
    }
}
